File: Basic Strategy Practice.py
from random import randint

ls = ["2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "K", "Q", "A"]
print("No.".ljust(4, " "), "Player".ljust(9, " "), "Dealer".ljust(9, " "), "Prediction")
print(36 * "-")
n, flag = 1, True
while True:
  print(str(n).ljust(5, " "), end="")
  if flag:
    print((ls[randint(0, 11)] + " " + ls[randint(0, 11)]).ljust(10, " "), end="") # Hard
  else:
    print(("A " + ls[randint(0, 7)]).ljust(10, " "), end="") # Soft
  # randPair = randint(0, 12) # Pair
  # print((ls[randPair] + " " + ls[randPair]).ljust(10, " "), end="") # Pair
  # print((ls[randint(0, 12)] + " " + ls[randint(0, 12)]).ljust(10, " "), end="") # Both
  print(ls[randint(0, 12)].ljust(10, " "), end="")
  input()
  n += 1
  flag ^= True

File: Basic Strategy Practice.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'

int main(void) {
	srand(time(NULL));

	vector<string> v = {"2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "K", "Q", "A"};
	string prediction;
	int n = 0;
	cout << left << setw(5) << "No." << setw(10) << "Player" << setw(10) << "Dealer" << setw(14) << "Prediction" << endl;
	cout << "--------------------------------------" << endl;
	bool flag = true;
	while (true) {
		cout << setw(5) << ++n;
		if (flag) {
			cout << setw(10) << v[(rand() % 12)] + " " + v[(rand() % 12)];	// Hard
		} else {
			cout << setw(10) << ("A " + v[(rand() % 8)]);	// Soft
		}
		// cout << setw(10) << v[(rand() % 13)] + " " + v[(rand() % 13)];	// Both
		cout << setw(10) << v[(rand() % 13)];
		cin >> prediction;
		flag ^= true;
	}
}
